src/components/molecules/Visualizer/hooks.ts

```diff
@@ -65,6 +69,33 @@ export default ({
     sceneProperty ?? {},
   );
 
+  const hasTimelineBuiltinWidget = useMemo(() => {
+    if (!alignSystem) {
+      return false;
+    }
+    return !!Object.keys(alignSystem).some(key => {
+      const align = alignSystem[key as keyof WidgetAlignSystemType];
+      if (!align) {
+        return false;
+      }
+      return !!Object.keys(align).some(alignKey => {
+        const pos = align[alignKey as keyof typeof align];
+        if (!pos) {
+          return false;
+        }
+        return !!Object.keys(pos).some(posKey => {
+          const widget = pos[posKey as keyof typeof pos];
+          if (!widget) {
+            return false;
+          }
+          return !!widget.widgets?.some(
+            w => `${w.pluginId}/${w.extensionId}` === TIMELINE_BUILTIN_WIDGET_ID,
+          );
+        });
+      });
+    });
+  }, [alignSystem]);
```

Comment:
I want to check if timeline widget is displayed but is there a smarter way to do this?
This is used for `requestRenderMode` and `maximumRenderTimeChange` props in Cesium component.
This is because, if we don't set these props and we operate timeline widget then cesium does not render earth by operating timeline widget.

Reply:
I'm sure @rot1024 has a better solution than what I'll suggest, but I think checking the widget [here](https://github.com/reearth/reearth-web/blob/7eead9b459f562b8b4c21f4cf571311d9c1c25af/src/components/organisms/EarthEditor/CanvasArea/convert.ts#L160) where we set the widgets you should do this. Do like a `let hasTimelineBuiltinWidget: boolean;` or useState and then inside the `convertWidgets` set it to true if there is a timeline widget

Reply:
I fixed your review in [refactor: add ownBuiltinWidgets property to convertWidgets](https://github.com/reearth/reearth-web/pull/285/commits/af54651e818dc7c529e103862cd5252646a59d61).
Thank you for your advice!!
And this commit fixes the following your comment.
- https://github.com/reearth/reearth-web/pull/285#discussion_r934156880
- https://github.com/reearth/reearth-web/pull/285#discussion_r934160698